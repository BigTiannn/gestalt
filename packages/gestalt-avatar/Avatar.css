:root {
  /*** Avatar Sizing ***/

  /* small screens */

  --sm-size-xs: 24px;
  --sm-size-sm: 40px;
  --sm-size-md: 72px;
  --sm-size-lg: 104px;
  --sm-size-xl: 132px;

  /* add medium screens when design adds third breakpoint option for avatars */

  /* large screens */

  --lg-size-xs: 36px;
  --lg-size-sm: 60px;
  --lg-size-md: 108px;
  --lg-size-lg: 156px;
  --lg-size-xl: 198px;
}

.defaultAvatar {
  composes: bg-gray from "gestalt-colors/Colors.css";
  composes: relative flex items-center justify-center from "gestalt-layout/Layout.css";
  composes: circle from "gestalt-borders/Borders.css";
}

.initial {
  composes: antialiased bold from "gestalt-typography/Typography.css";
  composes: white from "gestalt-colors/Colors.css";
}

.responsive {
  composes: col-12 from "gestalt-column/Column.css";
  padding-bottom: 100%;
}

.xs {
  height: var(--sm-size-xs);
  width: var(--sm-size-xs);
}

.sm {
  height: var(--sm-size-sm);
  width: var(--sm-size-sm);
}

.md {
  height: var(--sm-size-md);
  width: var(--sm-size-md);
}

.lg {
  height: var(--sm-size-lg);
  width: var(--sm-size-lg);
}

.xl {
  height: var(--sm-size-xl);
  width: var(--sm-size-xl);
}

@media (--lg) {
  .xs {
    height: var(--lg-size-xs);
    width: var(--lg-size-xs);
  }

  .sm {
    height: var(--lg-size-sm);
    width: var(--lg-size-sm);
  }

  .md {
    height: var(--lg-size-md);
    width: var(--lg-size-md);
  }

  .lg {
    height: var(--lg-size-lg);
    width: var(--lg-size-lg);
  }

  .xl {
    height: var(--lg-size-xl);
    width: var(--lg-size-xl);
  }
}
