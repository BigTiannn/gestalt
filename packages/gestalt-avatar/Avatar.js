// @flow
import React, { PropTypes } from 'react';
import classnames from 'classnames/bind';
import Heading from 'gestalt-heading';
import Image from 'gestalt-image';
import Mask from 'gestalt-mask';
import styles from './Avatar.css';

const cx = classnames.bind(styles);

type DefaultAvatarProps = {
  initial?: string,
  name: string,
  size: 'xs' | 's' | 'm' | 'l' | 'xl',
};

function DefaultAvatar(props: DefaultAvatarProps) {
  const { initial, name, size } = props;
  const firstInitial = initial || name.charAt(0).toUpperCase();
  const classes = cx(size, 'defaultAvatar');

  return (
    <div aria-label={name} className={classes}>
      <Heading size={size}>{firstInitial}</Heading>
    </div>
  );
}

type AvatarProps = {
  initial?: string,
  name: string,
  size: 'xs' | 's' | 'm' | 'l' | 'xl',
  src?: string,
  wash?: bool,
}

export default function Avatar(props: AvatarProps) {
  const {
    initial,
    name,
    size,
    src,
    wash = false,
  } = props;

  if (!src) {
    return (
      <DefaultAvatar initial={initial} name={name} size={size} />
    );
  }

  const classes = cx(size);

  return (
    <div className={classes}>
      <Mask shape="circle">
        <Image
          alt={name}
          color={'#efefef'}
          height={1}
          src={src}
          wash={wash}
          width={1}
        />
      </Mask>
    </div>
  );
}

Avatar.propTypes = {
  initial: PropTypes.string,
  name: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl']).isRequired,
  src: PropTypes.string,
  wash: PropTypes.bool,
};
