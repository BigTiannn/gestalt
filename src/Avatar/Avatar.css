:root {
  --styles: '../../styles.css';

  /* small screens */

  --sm-size-xs: 18px;
  --sm-size-s: 30px;
  --sm-size-m: 54px;
  --sm-size-l: 78px;
  --sm-size-xl: 99px;

  /* medium screens */

  --md-size-xs: 24px;
  --md-size-s: 40px;
  --md-size-m: 64px;
  --md-size-l: 104px;
  --md-size-xl: 132px;

  /* large screens */

  --lg-size-xs: 36px;
  --lg-size-s: 60px;
  --lg-size-m: 108px;
  --lg-size-l: 156px;
  --lg-size-xl: 198px;
}

.defaultAvatar {
  composes: bg-light-gray from var(--styles);
  composes: circle from var(--styles);
  composes: flex from var(--styles);
  composes: items-center from var(--styles);
  composes: justify-center from var(--styles);
}

@media (--sm) {
  .xs {
    height: var(--sm-size-xs);
    width: var(--sm-size-xs);
  }

  .s {
    height: var(--sm-size-s);
    width: var(--sm-size-s);
  }

  .m {
    height: var(--sm-size-m);
    width: var(--sm-size-m);
  }

  .l {
    height: var(--sm-size-l);
    width: var(--sm-size-l);
  }

  .xl {
    height: var(--sm-size-xl);
    width: var(--sm-size-xl);
  }
}

@media (--md) {
  .xs {
    height: var(--md-size-xs);
    width: var(--md-size-xs);
  }

  .s {
    height: var(--md-size-s);
    width: var(--md-size-s);
  }

  .m {
    height: var(--md-size-m);
    width: var(--md-size-m);
  }

  .l {
    height: var(--md-size-l);
    width: var(--md-size-l);
  }

  .xl {
    height: var(--md-size-xl);
    width: var(--md-size-xl);
  }
}

@media (--lg) {
  .xs {
    height: var(--lg-size-xs);
    width: var(--lg-size-xs);
  }

  .s {
    height: var(--lg-size-s);
    width: var(--lg-size-s);
  }

  .m {
    height: var(--lg-size-m);
    width: var(--lg-size-m);
  }

  .l {
    height: var(--lg-size-l);
    width: var(--lg-size-l);
  }

  .xl {
    height: var(--lg-size-xl);
    width: var(--lg-size-xl);
  }
}
