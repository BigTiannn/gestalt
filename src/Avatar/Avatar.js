import React, { PropTypes } from 'react';
import Image from '../Image/Image';
import Mask from '../Mask/Mask';

export default function Avatar(props) {
  const {
    avatar: {
      size,
      url,
    },
    name,
  } = props;
  return (
    <Mask height={size} type="circle" width={size}>
      <Image
        alt={name}
        color="#EFEFEF"
        height={size}
        src={url}
        width={size}
      />
    </Mask>
  );
}

Avatar.propTypes = {
  avatar: PropTypes.shape({
    size: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired,
  }).isRequired,
  name: PropTypes.string.isRequired,
};
