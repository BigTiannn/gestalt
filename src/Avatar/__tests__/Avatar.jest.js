/* eslint-env jest */
/* eslint import/imports-first: 0 */
jest.unmock('../Avatar');
jest.unmock('../../Box/Box');
jest.unmock('../../Image/Image');
jest.unmock('../../Mask/Mask');

import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '../Avatar';

test('Avatar renders multibyte names', () => {
  const component = renderer.create(
    <Avatar name="ðŸ’© astral" />,
    {
      createNodeMock() {
        return { clientWidth: 100 };
      },
    }
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
