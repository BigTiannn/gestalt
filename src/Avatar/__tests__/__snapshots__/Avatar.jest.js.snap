exports[`test Avatar renders multibyte names 1`] = `
<div
  className="col-12">
  <div
    aria-label="ğŸ’© astral"
    className="box xs-flex items-center justify-center bg-gray circle"
    postition="relative"
    style={
      Object {
        "fontSize": 50,
      }
    }>
    <div
      className="box relative"
      style={
        Object {
          "paddingBottom": "100%",
        }
      } />
    <div
      className="box">
      <span
        className="white bold antialiased">
        ğŸ’©
      </span>
    </div>
  </div>
</div>
`;
